packages/framework-core/test/services/read-model-store.test.ts

```diff
@@ -89,6 +97,23 @@ describe('ReadModelStore', () => {
       })
     })
 
+    context('when the new read model returns the deleteReadModel type', () => {
+      it('deletes the associated read model', async () => {
+        replace(config.provider.readModels, 'store', fake())
+        replace(config.provider.readModels, 'deleteReadModel', fake())
+        replace(
+          ReadModelStore.prototype,
+          'reducerForProjection',
+          fake.returns(() => deleteReadModel)
+        )
+        const readModelStore = new ReadModelStore(config, logger)
+
+        await readModelStore.project(anEntitySnapshot)
+        expect(config.provider.readModels.store).not.to.have.been.called
+        expect(config.provider.readModels.deleteReadModel).to.have.been.calledTwice
```

Comment:
Why twice?

Reply:
hey @NickSeagull `anEntitySnapshot` has two projections associated, I'm deleting both. Maybe is it better to just mock one? 🤔

Reply:
I was just asking :) its okay! :)